### PR TITLE
Using pid as a unique identifier of the ws session processes

### PR DESCRIPTION
**Pull request by shamis**
encodes the pid to a binary and uses as using ets had problems when working on a cluster

### CODE REVIEW

File: src/socketio_session.erl [hunk 1]
```diff
@@ -50,19 +47,16 @@ configure(Opts) ->
...
-    Pid.
+create(SessionTimeout, Callback, Opts) ->
+    {ok, Pid} = socketio_session_sup:start_child(SessionTimeout, Callback, Opts),
+    base64:encode(term_to_binary(Pid)).
 
-find(SessionId) ->
-    case global:whereis_name(SessionId) of
+find(PidBin) ->
+    Pid = binary_to_term(base64:decode(PidBin)),
+    case process_info(Pid) of
```
Comment (c-bik) [inline src/socketio_session.erl]: Can you use `erlang:is_process_alive/1` instead?

File: src/socketio_session.erl [hunk 2]
```diff
@@ -178,22 +171,17 @@ handle_cast(_Msg, State) ->
...
-            case Callback:open(self(), SessionId, Opts) of
-                {ok, SessionState} ->
-                    send(self(), {connect, <<>>}),
-                    {noreply, State#state{registered = true, session_state = SessionState}};
-                disconnect ->
-                    {stop, normal, State}
-            end;
-        false ->
-            {stop, session_id_exists, State}
+handle_info(register_in_ets, State = #state{registered = false, callback = Callback, opts = Opts}) ->
```
Comment (c-bik) [inline src/socketio_session.erl]: Why is this still needed?

File: src/socketio_session.erl (hunk 1 above)
Comment (shamis) [inline src/socketio_session.erl]: I think it works with only local pids

**Comment by shamis** [comment]
@c-bik have made the changes please verify

File: src/socketio_session.erl (hunk 1 above)
Comment (c-bik) [inline src/socketio_session.erl]: `process_info` also doesn't work. Use:

``` erlang
rpc:call(erlang:node(P), erlang, is_process_alive, [P]).
```

Comment (shamis) [inline src/socketio_session.erl]: have made the change. please verify

**Pull request merged**